### ISSUE TITLE
Inconsistent AST formatting for subqueries with tuples

### ISSUE THREAD

**Issue by antonio2368** [ClickHouse/ClickHouse#62227]
https://s3.amazonaws.com/clickhouse-test-reports/62212/a7c490e1dfeadc6f968a57aca147683663fbe766/stateless_tests__debug__s3_storage__[5_6].html

```
<Fatal> : Logical error: 'Inconsistent AST formatting: the query:
SELECT NOT (SELECT NULL, CAST([], 'String'))
cannot parse.'.
```

This was the original query:
```sql
SELECT NOT ((SELECT NULL, CAST([], 'String')))
```

The problem is that we remove one set of `( )`.

**Comment by vitlibar** [comment]
I came across another reproduction of this bug (https://s3.amazonaws.com/clickhouse-test-reports/66061/87dda31a2c234a7596b661ad5d63f74f3124ea25/ast_fuzzer__debug_.html):
```
CREATE TABLE tbl (x Bool) ENGINE=MergeTree ORDER BY tuple();
ALTER TABLE tbl MODIFY COLUMN x Nullable(UInt8, tuple(1, toLowCardinality(2)));
```

Here `Nullable(UInt8, tuple(1, toLowCardinality(2)))` is an invalid expression (`Nullable` must have only one argument), but the server stops with `Logical error: 'Inconsistent AST formatting` instead of showing an error message.

**Comment by nickitat** [comment]
``` sql
SELECT *, e, e2 FROM t1 FULL OUTER JOIN t2 ON (((t1.x = t2.x) AS e) AND ((t2.x IS NOT NULL) AND (t1.x IS NOT NULL))) OR ((t2.x IS NULL) AND (t1.x IS NULL)) AS e2 ORDER BY t1.x ASC NULLS LAST, t2.x ASC NULLS LAST
```

https://s3.amazonaws.com/clickhouse-test-reports/66336/6ffac4034a4ca1b18a2ad5aa44f2c7cce696c246/stateless_tests__debug__s3_storage__[2_2]/clickhouse-server.err.log

**Comment by azat** [comment]
@nickitat this is different, it will be fixed in https://github.com/ClickHouse/ClickHouse/pull/66378

**Comment by alexey-milovidov** [comment]
I fixed it in #66840.
